Homework 22

ElenaChurinova opened the pull request:
Implement the POM for a "RenamePlaylist()" test.




I have a question , why my previous homeworks (#19,#20,#21) are attached as well? How I can exclude them from my pull request?
I tried to deleted them from the InteliJ IDIA but it hasn't changed anything. 
Thank you!!!!

ElenaChurinova committed:
Homework19

Verify that the notification displayed has the text, "Deleted playlist {playlist name}".

ElenaChurinova committed:
Homework20

Removed all implementations of Thread.sleep. and used  Explicit Waits instead

ElenaChurinova committed:
Homework21

ElenaChurinova committed:
Homework22

Berichicko commented on src/test/java/Pages/BasePage.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.Duration;
+
+public class BasePage {
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
```
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
    protected Actions actions;
    protected WebDriverWait wait;
    protected WebDriver driver;
```

Berichicko commented on src/test/java/Pages/HomePage.java:
```diff
@@ -0,0 +1,25 @@
+        super(givenDriver);
+    }
+    // Web Elements:
+    By playlistElement = By.cssSelector("#playlists > ul > li:nth-child(3)");
```
You should use encapsulation of non-static and non-final fields.

If a variable is used only by other methods of that class, it should be declared as private.

I recommend:  make all the locators and WebElements private, because private keyword in java allows most restrictive access to variables and methods and offer strongest form of Encapsulation. 

For example

    ```
    private By shuffleBtnLocator = By.cssSelector(".btn-shuffle-all");
    private By firstSongLocator = By.cssSelector(".play");

```

Berichicko commented on src/test/java/Pages/LoginPage.java:
```diff
@@ -0,0 +1,34 @@
+        super(givenDriver);
+    }
+    //WEb Elements:
+    By emailField = By.cssSelector("input[type='email']");
+    By passwordField = By.cssSelector("input[type='password']");
+    By clickBtn = By.cssSelector("button[type='submit']");
```
You should use encapsulation of non-static and non-final fields.

If a variable is used only by other methods of that class, it should be declared as private.

I recommend:  make all the locators and WebElements private, because private keyword in java allows most restrictive access to variables and methods and offer strongest form of Encapsulation. 

For example

    ```
    private By shuffleBtnLocator = By.cssSelector(".btn-shuffle-all");
    private By firstSongLocator = By.cssSelector(".play");

```

Berichicko commented on src/test/java/Homework22.java:
```diff
@@ -0,0 +1,51 @@
+
+    }
+
+      //public void doubleClickPlaylist() {
+        //WebElement playlistElement = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("#playlists > ul > li:nth-child(3)")));
+        //actions.doubleClick(playlistElement).perform();
+    //}
+    //public void enterNewPlaylistName(){
+        //WebElement playListInputField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[name='name']")));
+        //playListInputField.sendKeys(Keys.chord(Keys.COMMAND, "A", Keys.BACK_SPACE));
+        //playListInputField.sendKeys(newPlaylistName);
+        //playListInputField.sendKeys(Keys.ENTER);
+   // }
+    //public String expectedSuccessMsg(){
+     //  WebElement successMsg = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.success.show")));
+     //  return successMsg.getText();
+   // }
```
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.

Berichicko commented on src/test/java/Pages/BasePage.java:
```diff
@@ -0,0 +1,27 @@
+package Pages;
```
Package names are written in all lower case to avoid conflict with the names of classes or interfaces.

For example `pages`
Rename refactorings https://www.jetbrains.com/help/idea/rename-refactorings.html#rename_class_example

Berichicko commented on src/test/java/Homework22.java:
```diff
@@ -0,0 +1,51 @@
+
+public class Homework22 extends BaseTest{
+
+    @Test
+public void testRenamePlaylist()  {
```
How can I beautify code in IntelliJ:

The formatting shortcuts in Intellij IDEA are :
For Windows : Ctrl + Alt + L.
For Ubuntu : Ctrl + Alt + Windows + L.
For Mac : ⌥ (Option) + ⌘ (Command) + L.

![image](https://user-images.githubusercontent.com/104761482/209774134-9e63bfed-d768-4455-a860-028305c85e17.png)

Berichicko commented:
> Implement the POM for a "RenamePlaylist()" test.
> 
> I have a question , why my previous homeworks (#19,#20,#21) are attached as well? How I can exclude them from my pull request? I tried to deleted them from the InteliJ IDIA but it hasn't changed anything. Thank you!!!!

Hi Elena

Try it

Creating and deleting branches within your repository https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository

https://www.youtube.com/watch?v=A8MUpT9WOd0

https://www.git-tower.com/learn/git/faq/github-delete-branch

![image](https://github.com/user-attachments/assets/5d591704-ba9e-43df-a969-242f9f32d9c3)

ElenaChurinova committed:
Homework22

Deleted unused comments, made all the locators and WebElements private, made the variables protected and changed the package name in all lower case.

ElenaChurinova committed:
Homework22

Berichicko commented:
Hello,

I have checked your homework 22, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG](https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b)

![Quote](https://github-readme-quotes-bay.vercel.app/quote?theme=dark&layout=socrates)


![image](https://github.com/user-attachments/assets/4429b21d-c468-4f4f-a62f-029f0c9bdd69)
